terraform-modules/aws/msk/main.tf

```diff
@@ -0,0 +1,164 @@
+# │       status code: 400, request id: 3ba26851-f96a-48b6-a9a2-ca7a68be8e5f
+# │ 
+# │   with aws_acmpca_certificate_authority.this,
+# │   on main.tf line 91, in resource "aws_acmpca_certificate_authority" "this":
+# │   91: resource "aws_acmpca_certificate_authority" "this" {
+# resource "aws_s3_bucket_server_side_encryption_configuration" "this" {
+#   bucket = aws_s3_bucket.this.bucket
+
+#   rule {
+#     apply_server_side_encryption_by_default {
+#       kms_master_key_id = aws_kms_key.this.arn
+#       sse_algorithm     = "aws:kms"
+#     }
+#   }
+# }
+
+data "aws_iam_policy_document" "acmpca_bucket_access" {
+  count = var.s3_bucket_create ? 1 : 0
+  statement {
+    actions = [
+      "s3:GetBucketAcl",
+      "s3:GetBucketLocation",
+      "s3:PutObject",
+      "s3:PutObjectAcl",
+    ]
+
+    resources = [
+      aws_s3_bucket.this[0].arn,
+      "${aws_s3_bucket.this[0].arn}/*",
+    ]
+
+    principals {
+      identifiers = ["acm-pca.amazonaws.com"]
+      type        = "Service"
+    }
+  }
+}
+
+resource "aws_s3_bucket_policy" "this" {
+  count = var.s3_bucket_create ? 1 : 0
+  bucket = aws_s3_bucket.this[0].id
+  policy = data.aws_iam_policy_document.acmpca_bucket_access[0].json
+}
+
+#######################################
+# Private CA
+#######################################
+data "aws_partition" "current" {
+}
+
+resource "aws_acmpca_certificate_authority_certificate" "cacert" {
+  certificate_authority_arn = aws_acmpca_certificate_authority.this.arn
+
+  certificate       = aws_acmpca_certificate.cert.certificate
+  certificate_chain = aws_acmpca_certificate.cert.certificate_chain
+}
+
+resource "aws_acmpca_certificate" "cert" {
+  certificate_authority_arn   = aws_acmpca_certificate_authority.this.arn
+  certificate_signing_request = aws_acmpca_certificate_authority.this.certificate_signing_request
+  signing_algorithm           = "SHA512WITHRSA"
+
+  template_arn = "arn:${data.aws_partition.current.partition}:acm-pca:::template/RootCACertificate/V1"
+
+  validity {
+    type  = "YEARS"
+    value = local.years_valid
+  }
+}
+
+resource "aws_acmpca_certificate_authority" "this" {
+  certificate_authority_configuration {
+    key_algorithm     = var.key_algorithm
+    signing_algorithm = var.signing_algorithm
+
+    subject {
+      common_name = var.common_name
+    }
+  }
+  
+  type = "ROOT"
+
+  revocation_configuration {
+    crl_configuration {
+      custom_cname       = "crl.${var.common_name}"
+      enabled            = true
+      expiration_in_days = var.expiration_in_days
+      s3_bucket_name     = var.s3_logs_bucket
+    }
+  }
+  
+  tags   = var.tags
+
+  depends_on = [aws_s3_bucket_policy.this]
+}
+
+#######################################
+# MSK Cluster
+#######################################
+module "msk" {
+  source                         = "cloudposse/msk-apache-kafka-cluster/aws"
+  version                        = "v0.8.4"
```

Comment:
should we update to the latest? https://github.com/cloudposse/terraform-aws-msk-apache-kafka-cluster

Reply:
yes it was updated to lastest
but we are pointing to a branch : https://github.com/cloudposse/terraform-aws-msk-apache-kafka-cluster/pull/76